integration-tests/tests/run-analysis-azure.ts

```diff
@@ -69,8 +71,60 @@ const testRunAnalysisAzure = _.flowRight(
   await findElement(page, clickable({ textContains: 'JupyterLab Environment' }));
   await findElement(page, clickable({ textContains: 'Creating' }));
 
-  // Wait for env to finish creating, or break early on error
-  await findElement(page, clickable({ textContains: 'Running' }), { timeout: Millis.ofMinutes(25) });
+  // Wait for env to finish creating, or break early on only errors related to runtime creation
+  await retryUntil({
+    getResult: async () => {
+      await dismissInfoNotifications(page);
+
+      // check for errors
+      const hasError = await openError(page);
+      if (hasError) {
+        let hasfatalError = false;
+        // Occurs if leonardo creation call itself errors
+        try {
+          await findText(page, 'Error modifying cloud environment', { timeout: Millis.ofSecond });
+          // Prior line will throw and go to catch block if not found
+          hasfatalError = true;
+        } catch {}
+
+        // Occurs if the runtime transitions to error status
+        try {
+          await findText(page, 'Cloud Environment is in error state', { timeout: Millis.ofSecond });
+          // Prior line will throw and go to catch block if not found
+          hasfatalError = true;
+        } catch {}
+
+        if (hasfatalError) {
+          throw new Error('Failed to create cloud environment');
+        } else {
+          console.warn(
+            'Discovered an error in `run-analysis-azure` not related to runtime creation. Worth investigation'
+          );
+        }
+        // Dismiss any non-fatal errors
+        await dismissAllNotifications(page);
+      }
+
+      // Ensure any secondary notifications are dismissed as well
+      await dismissAllNotifications(page);
+
+      // Check for runtime running. findElement returns an error if it does not find the element, so we try/catch
+      try {
+        // Timeout fast here, the retry loop is the code block responsible for waiting
+        await findElement(page, clickable({ textContains: 'Running' }, { timeout: Millis.ofSecond }));
+        // True exits the loop.
+        // If we reach this line, findElement did not throw an exception, meaning we did indeed find `Running`.
+        return true;
+      } catch {
+        // If we catch an exception, we could not find `Running`, and want to loop again
+        console.log('Looping again in retryUntil for run-analysis-azure test');
+        return false;
+      }
+    },
+    interval: Millis.ofSeconds(30),
+    leading: true,
```

Comment:
Generally this logic looks sound, but what does this `leading` field do?

Reply:
Whether or not to wait the delay interval prior to the first loop (here, we don't and check if there are errors right away)

Reply:
ah ok, probably unlikely there's an error that quick, but no real harm in checking